bundler/spec/realworld/edgecases_spec.rb

```diff
@@ -450,10 +450,13 @@ def rubygems_version(name, requirement)
 
     bundle :lock, :env => { "DEBUG_RESOLVER" => "1" }
 
+    expect(out).to include("BUNDLER: Unactivated: \n")
+    expect(out).to include("BUNDLER: Finished resolution (")
+    steps = out.match(/BUNDLER: Finished resolution \((\d+) steps\)/)&.[](1)&.to_i
     if Bundler.feature_flag.bundler_3_mode?
-      expect(out).to include("BUNDLER: Finished resolution (2492 steps)")
+      expect(steps).to be_between(2420, 2520)
     else
-      expect(out).to include("BUNDLER: Finished resolution (2722 steps)")
+      expect(steps).to be_between(2650, 2750)
```

Comment:
Why is this change? Is it because the number of resolution steps in the spec is now non-deterministic somehow, or because you want to make the spec a bit more resilient to future changes in the resolver? If it's the latter, resolver's code hardly ever changes, so I don't mind having to update this spec when changes we make affect the number of steps needed. Actually, the performance spec you added is essentially testing the same thing as here: that the resolver doesn't take too long, so we can probably keep both tests but unify the approach.

Reply:
On my mac on the `master` using ruby 2.7.2 this test failed because the number of steps were different. So it appears to be deterministic on github actions in exactly the case it is run in, but only then. Given that, it felt like a fragile test. With a range, this test could also be opened up beyond just ruby 2.7.2. 

Here is the beginning of the very long output:

```sh
$ bin/rspec spec/realworld/edgecases_spec.rb:349 --tag realworld                                                                            (master %=)
Run options:
  include {:realworld=>true, :locations=>{"./spec/realworld/edgecases_spec.rb"=>[349]}}
  exclude {:jruby=>true, :readline=>false, :permissions=>false, :no_color_tty=>false, :ruby_repo=>false, :bundler=>"!= 2", :git=>"!= 2.30.0", :sudo=>true}

Randomized with seed 16292
F

Retried examples: 1
  real world edgecases doesn't hang on big gemfile
  [5/5] ./spec/realworld/edgecases_spec.rb:349


Failures:

  1) real world edgecases doesn't hang on big gemfile
     Failure/Error: expect(out).to include("BUNDLER: Finished resolution (2722 steps)")

       expected "Starting resolution (2021-01-04 17:54:33 +0000)\nUser-requested dependencies: [#<Gem::Resolver::Depe...Writing lockfile to /Users/lukas.oberhuber/projects/rubygems/bundler/tmp/1/bundled_app/Gemfile.lock" to include "BUNDLER: Finished resolution (2722 steps)"
       Diff:
       @@ -1,10033 +1,20065 @@
       -BUNDLER: Finished resolution (2722 steps)
       +Starting resolution (2021-01-04 17:54:33 +0000)
```

Way further down:

```sh
BUNDLER: Finished resolution (2738 steps) (Took 10.813019 seconds) (2021-01-04 17:54:47 +0000)
```

Reply:
I see. How about changing to the same strategy you used for the test you added, just making sure it doesn't take too long?

Reply:
Actually, it seems fine as it is, maybe leave a note about this for future reference.

Reply:
Will do.

Reply:
Done in 8574da2823d1c9b11f0e5bab69a5dd597f15beb8

Reply:
:+1:

Reply:
Do you have any idea why the number of steps now changed? I'm refering to https://github.com/rubygems/rubygems/pull/4216/commits/acc478ddbb031236740ae11506fb7cb040c9cdb1.

Anyways, I'm think that maybe it's better to keep the exact number of steps here. It makes us more aware of subtle changes in the resolver. I think we don't need to open these specs to other rubies, they represent specific realworld cases so it's fine to run the exact scenario of the original report.  And regarding running specs locally on MacOS (which is what prompted you to change this),I'm planning to check it in CI (see #4245), so it will be fixed by that :+1:.

Anyways, I realize I'm probably just being annoying at this point, and I'm planning an imminent release, so I can take this over and make  the last tweaks if that's ok with you.

Reply:
> Anyways, I realize I'm probably just being annoying at this point, and I'm planning an imminent release, so I can take this over and make the last tweaks if that's ok with you.

Not annoying. All good comments. In any case, go for it. Happy for you to drive it to live :)

Reply:
Don't forget one of the changes is on the vendored `Molinillo`.

Reply:
Oh, of course, I forgot! Are you planning to propose them upstream? In any case, the changes we upstream to molinillo are usually very slow to get released, so what I normally do is include them and add a note to our "vendoring rake tasks" to keep track of the unreleased patches we are shipping.